src/transformers/utils/attention_visualizer.py

```diff
@@ -0,0 +1,203 @@
+            [1 if (0 <= i - j < sliding_window) or img_token in words[i] and img_token in words[j] else 0 for j in range(n)]
+            for i in range(n)
+        ]
+
+    row_dummy = " ".join(
+        f"{YELLOW}{BLACK_SQUARE}{RESET}"
+        if img_token in words[j] and mask[i, j] and img_token in words[i]
+        else f"{GREEN}{BLACK_SQUARE}{RESET}"
+        if i == j
+        else BLACK_SQUARE
+        if mask[0, j]
+        else WHITE_SQUARE
+        for j in range(n)
+    )
+    # Print headers
+    legend = f"{GREEN}{BLACK_SQUARE}{RESET}: i == j (diagonal)   {YELLOW}{BLACK_SQUARE}{RESET}: img tokens"
+    print(" " + legend)
+    f_string = " " * (max_word_length + 5) + "Attention Matrix".ljust(len(row_dummy))
+    if sliding_window is not None:
+        f_string+= "Sliding Window Mask"
+    print(f_string)
+    vertical_header = []
+    for idx, word in enumerate(words):
+        if img_token not in word:
+            vertical_header += [list(str(idx).rjust(len(str(n))))]
+        else:
+            vertical_header += [[f"{YELLOW}{k}{RESET}" for k in list(str(idx).rjust(len(str(n))))]]
+
+    vertical_header = list(map(list, zip(*vertical_header)))  # Transpose
+
+    for row in vertical_header:
+        print((max_word_length + 5) * " " + " ".join(row) + "    |    " + " ".join(row) if sliding_window is not None else "")
+
+    for i, word in enumerate(words):
+        word_repr = repr(word).ljust(max_word_length)
+        colored_word = f"{YELLOW}{word_repr}{RESET}" if img_token in word else word_repr
+        row_display = " ".join(
+            f"{YELLOW}{BLACK_SQUARE}{RESET}"
+            if img_token in words[j] and mask[i, j] and img_token in words[i]
+            else f"{GREEN}{BLACK_SQUARE}{RESET}"
+            if i == j
+            else BLACK_SQUARE
+            if mask[i, j]
+            else WHITE_SQUARE
+            for j in range(n)
+        )
+        sliding_window_row = ""
+        if sliding_window is not None:
+            sliding_window_row = " ".join(
+                f"{YELLOW}{BLACK_SQUARE}{RESET}"
+                if img_token in words[j] and sliding_window_mask[i][j] and img_token in words[i]
+                else f"{GREEN}{BLACK_SQUARE}{RESET}"
+                if i == j
+                else BLACK_SQUARE
+                if sliding_window_mask[i][j]
+                else WHITE_SQUARE
+                for j in range(n)
+            )
+
+        print(f"{colored_word}: {str(i).rjust(2)} {row_display}    |    {sliding_window_row}")
+
+
+class AttentionMaskVisualizer:
+    def __init__(self, model_name: str):
+        config = AutoConfig.from_pretrained(model_name)
+        self.image_token = "<img>"
+        if hasattr(config.get_text_config(), "sliding_window"):
+            config.sliding_window = 5
+        try:
+            mapped_cls = _get_model_class(config, MODEL_MAPPING)
+        except Exception:
+            mapped_cls = _get_model_class(config, MODEL_FOR_PRETRAINING_MAPPING)
```

Comment:
maybe a clearer exception handling here? (added e but not sure what to do)
```suggestion
        try:
            mapped_cls = _get_model_class(config, MODEL_MAPPING)
        except Exception as e:
            mapped_cls = _get_model_class(config, MODEL_FOR_PRETRAINING_MAPPING)
```